homeassistant/components/intent/__init__.py

```diff
@@ -89,10 +101,22 @@ async def async_setup_intents(self, hass: HomeAssistant) -> None:
 class OnOffIntentHandler(intent.ServiceIntentHandler):
     """Intent handler for on/off that also supports covers, valves, locks, etc."""
 
+    def __init__(
+        self,
+        intent_type: str,
+        domain: str,
+        service: str,
+    ) -> None:
+        """Create on/off intent handler."""
+        super().__init__(intent_type, domain, service)
+
```

Comment:
This does nothing.
```suggestion
```

Reply:
Oh nevermind, it's changing the type.

Reply:
The `service` parameter here is `str` as opposed to `str | None` in the base class because the service name is always known ahead of time for on/off.